soc/arm/microchip_mec/common/pinctrl_soc.h

```diff
@@ -26,71 +26,30 @@ extern "C" {
 /** @cond INTERNAL_HIDDEN */
 
 /* Type for MCHP XEC pin. */
-typedef struct {
-	/** Pinmux settings (port, pin and function). */
-	uint16_t pinmux;
-	/** Pin configuration (bias, drive and slew rate). */
-	uint16_t pincfg;
-} pinctrl_soc_pin_t;
+typedef uint32_t pinctrl_soc_pin_t;
```

Comment:
This will cause backward compatibility break for all code using existing properties.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/16172078/6b48d096-4372-4a91-8df0-a0fead156bea)

Reply:
e.g. const pinctrl_soc_pin_t pin = { MCHP_XEC_PINMUX(060, MCHP_AF2), 0 }; /* set pinmux and pincfg */
does this now becomes simply? Where do the pincfg gets applied? 
const pinctrl_soc_pin_t pin = MCHP_XEC_PINMUX(060, MCHP_AF2)

Reply:
Yes, you are correct.
For "const pinctrl_soc_pin_t pin = { MCHP_XEC_PINMUX(060, MCHP_AF2), 0 };" 
it become "const pinctrl_soc_pin_t pin = MCHP_XEC_PINMUX(060, MCHP_AF2)"
Since we have 0 as pincfg, we no need to apply pincfg explicitly.

Some pincfg usage:
1) To set Pull Up on the pin 060:
	const pinctrl_soc_pin_t pin = MCHP_XEC_PINMUX(060, MCHP_AF1) | (1u << MCHP_XEC_PU_POS);
2) To set 2x drive strength on the pin 060:
	const pinctrl_soc_pin_t pin = MCHP_XEC_PINMUX(060, MCHP_AF1) | (MCHP_XEC_DRV_STR0_2X << MCHP_XEC_DRV_STR_POS);
	
The pincfg macro is defined in 
	include/zephyr/dt-bindings/pinctrl/mchp-xec-pinctrl.h
The mapping for DTS is defined in
	soc/arm/microchip_mec/common/pinctrl_soc.h